addon/i3dio/ui/object.py

```diff
@@ -308,7 +312,35 @@ class I3DNodeObjectAttributes(bpy.types.PropertyGroup):
                     "Summer = 100 / "
                     "Summer + Sun = 101",
         default=i3d_map['weather_prevent_mask']['default']
-    )    
+    )
+
+    viewer_spaciality_required_mask: StringProperty(
+        name="Viewer Spaciality Required Mask (Hex)",
+        description="The Viewer Spaciality Required Mask as a hexadecimal value.",
+        default=i3d_map['viewer_spaciality_required_mask']['default']
+    )
+
+    viewer_spaciality_prevent_mask: StringProperty(
+        name="Viewer Spaciality Prevent Mask (Hex)",
+        description="The Viewer Spaciality Prevent Mask as a hexadecimal value.",
+        default=i3d_map['viewer_spaciality_prevent_mask']['default']
+    )
+
+    render_invisible: BoolProperty(
+        name="Render Invisible",
+        description='If set, the object is always rendered and "visibility"'
+        'must be controlled in the shader using the visible shader parameter',

```

Comment:
I assume this property is named this way from the way it is in GE? Because that is such a horrible property name given that it is in fact not invisible, but just shader controlled right? 😂

Reply:
Exactly 😅